core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -392,10 +397,36 @@ private[spark] class ExecutorAllocationManager(
   }
 
   /**
-   * Request the cluster manager to remove the given executor.
+   * Request the cluster manager to remove the given executors.
    * Return whether the request is received.
    */
-  private def removeExecutor(executorId: String): Boolean = synchronized {
+  private def removeExecutors(executorIds: Seq[String]): Boolean = synchronized {
+
+    val executorIdsToBeRemoved = executorIds.filter(canBeKilled)
+
+    // Send a request to the backend to kill this executor
+    val removeRequestAcknowledged = testing || client.killExecutors(executorIdsToBeRemoved)
```

Comment:
> Killing any one of them should return true

That's the current behavior, so it should be preserved. Yeah it's sketchy but there really isn't a good solution, so go with the current state of things.

> Can we change the api to return the list of killed executors

It's marked as `@DeveloperApi`, but even then, it's frowned upon to break compatibility for these APIs.